### DISCUSSION TITLE
Problems using [Route: filament.admin.resources.categories.edit] [URI: admin/categories/{record}/edit] [Missing parameter: record].

### DISCUSSION THREAD

**Discussion by StefanoTesla** [filamentphp/filament#10387] [Help]
I have a Category model that can have many Products that can have many attributes.

I created Category, Product and Attributes Resources.

I can see all the attributes, but every time I try to enter on Product and Category I get this error:

`Missing required parameter for [Route: filament.admin.resources.categories.edit] [URI: admin/categories/{record}/edit] [Missing parameter: record].`

This is the Category model:
```

<?php

namespace App\Models\Products;

use Illuminate\Database\Eloquent\Model;
use App\Models\Products\Product;
use Spatie\Translatable\HasTranslations;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Category extends Model implements \Mcamara\LaravelLocalization\Interfaces\LocalizedUrlRoutable
{
    use HasTranslations;

    public $translatable = ['name','description','slug'];
    public $timestamps = false;

    public function products(): HasMany
    {
        return $this->hasMany(Product::class);
    }

    public function getRouteKeyName()
    {
        $currentLocale = App()->getLocale();
        return "slug->{$currentLocale}";
    }

    public function getLocalizedRouteKey($locale)
    {
        return $this->getTranslation('slug',$locale);
    }

    public static function resolveBySlug($slug)
    {
        $currentLocale = App()->getLocale();
        return static::where("slug->{$currentLocale}", $slug);
    }


}

```

I'm using mcamara/laravel-localization and spatie/laravel-translatable and for some reason I need to use the `resolveBySlug` can this create some problems?

This is the final stack:

https://flareapp.io/share/Bm02jbe7#debug

Is it possible that the name in each product and category can create the problem cause they are json `{ 'en':'title,'it':'titolo'}`?? I already install the filament spatie translatable plugin

**Comment by anything-good** [comment]
any one has a solution to this one

**Comment by MarcelWeidum** [comment]
I think you should use `$this->record` instead of `$record` in your `Filament\Resources\Pages\ListRecords:156`
